### PR TITLE
issue-740

### PR DESCRIPTION
**Pull request by Jeff-Walker**
Adds null-safe EventType->Marker translation.
Adds unit test for new behavior.

fixes: https://github.com/ESAPI/esapi-java-legacy/issues/740

### CODE REVIEW

File: src/main/java/org/owasp/esapi/logging/slf4j/Slf4JLogBridgeImpl.java [hunk 1]
```diff
@@ -58,20 +58,46 @@ public void log(Logger logger, int esapiLevel, EventType type, String message) {
             throw new IllegalArgumentException("Unable to lookup SLF4J level mapping for esapi value of " + esapiLevel);
         }
         if (handler.isEnabled(logger)) {
+            type = safeEventType(type, esapiLevel);
```
Comment (jeremiahjstacey) [inline src/main/java/org/owasp/esapi/logging/slf4j/Slf4JLogBridgeImpl.java:61]: My opinion is that this should account for null, and only null. .

`type = type == null? org.owasp.esapi.Logger.EVENT_UNSPECIFIED : type;`

The additional types for ERROR and Fatal are  already being accounted for in the delegate Handler being called, and will be reflected in the level at which the SLF4J logger records the event.

I don't think the additional evaluation of the esapiLevel value is needed.

(borrowed from the SLF4JFactory class)
```
 Map<Integer, Slf4JLogLevelHandler> levelLookup = new HashMap<>();
        levelLookup.put(Logger.ALL, Slf4JLogLevelHandlers.TRACE);
        levelLookup.put(Logger.TRACE, Slf4JLogLevelHandlers.TRACE);
        levelLookup.put(Logger.DEBUG, Slf4JLogLevelHandlers.DEBUG);
        levelLookup.put(Logger.INFO, Slf4JLogLevelHandlers.INFO);
        levelLookup.put(Logger.ERROR, Slf4JLogLevelHandlers.ERROR);
        levelLookup.put(Logger.WARNING, Slf4JLogLevelHandlers.WARN);
        levelLookup.put(Logger.FATAL, Slf4JLogLevelHandlers.ERROR);
```

Comment (jeremiahjstacey) [reply src/main/java/org/owasp/esapi/logging/slf4j/Slf4JLogBridgeImpl.java:61]: Additionally, I do not think that null is handled in the JavaLogBridgeImpl either.  For consistency, I think it would be good to apply the ternary check in that class as well to prevent the same problem from being repeated in that scope.

Comment (Jeff-Walker) [reply src/main/java/org/owasp/esapi/logging/slf4j/Slf4JLogBridgeImpl.java:61]: Comment 1: that's fair. I'll remove and resubmit.
Comment 2: I'm pretty sure it does handle null, because my old code works. It's because the bridge impl doesn't do anything with the event type object but pass it to the appender, who handles the null no problem. The only reason that the SLF4J bridge has a problem is that it calls toString to make the marker object. I will go over it to make sure.

**Comment by Jeff-Walker** [comment]
I made the update. I also added a test to the JUL bridge to make sure it doesn't throw an exception on a null event type.

**Comment by Jeff-Walker** [comment]
Cool, thanks. 
Do you have a target date for the next release yet?

**Comment by kwwall** [comment]
@Jeff-Walker - TBH there hasn't been a lot done since the 2.5.0.0 release on 7/20/2022 and there is a significant time sink in doing the release (typically 2 hrs minimum plus whatever it takes to write up the release notes). Admittedly, the release notes would be comparatively simple this time. I would ideally like to knock out a few more bug fixes before the next release, so I think something around end of Oct / beginning of Nov would be more realistic.

**Comment by Jeff-Walker** [comment]
Thanks again.
